File: dato.config.rb

create_data_file "source/_data/homepage.yml", :yaml,
  site_name: dato.homepage.site_name,
  tag_line: dato.homepage.tag_line,
  description: dato.homepage.description

create_data_file "source/_data/seasons.yml", :yaml,
  (dato.seasons.map do |season|
    {
      title: season.name,
      image_url: season.image.file.width(400).to_url,
      thumbnail_url: season.image.file.height(300).to_url,
      bg_url: season.image.file.width(5).to_url,
      weight: season.position,
      overview: season.overview
    }
  end)

create_data_file "source/_data/episodes.yml", :yaml,
  (dato.episodes.sort_by { |e| [e.season.id.to_i, e.episode_number] }.map.with_index do |episode, i|
    {
      title: episode.title,
      episode_number: episode.episode_number,
      palette_url: episode.image && episode.image.file.auto("enhance").palette("json").to_url,
      image_url: episode.image && episode.image.file.width(500).to_url,
      thumbnail_url: episode.image && episode.image.file.width(500).height(280).fit("crop").auto("enhance").format(:jpg).to_url,
      date: episode.first_aired.to_s,
      weight: i,
      rating: episode.rating,
      director: episode.director,
      category: episode.season.name,
      description: episode.description
    }
  end)

create_data_file "source/_data/characters.yml", :yaml,
  (dato.characters.map do |character|
    {
      title: character.name,
      actor_name: character.actor_name,
      episodes: character.episode,
      weight: character.position,
      thumbnail_url: character.image.file.fit('crop').crop('faces').width(200).height(200).to_url,
      image_url: character.image.file.width(500).format(:jpg).to_url,
      description: character.description
    }
  end)

File: dato.config.js
const by = require('sort-by');

module.exports = (dato, root) => {
  root.createDataFile(
    'source/_data/homepage.yml', 'yaml',
    {
      siteName: dato.homepage.siteName,
      tagLine: dato.homepage.tagLine,
      description: dato.homepage.description,
    }
  );

  root.createDataFile(
    'source/_data/seasons.yml', 'yaml',
    dato.seasons.sort(by('name')).map(season => {
      return {
        title:        season.name,
        imageUrl:     season.image.url({ w: 400 }),
        thumbnailUrl: season.image.url({ h: 300 }),
        bgUrl:        season.image.url({ w: 5 }),
        weight:       season.position,
        overview:     season.overview
      };
    })
  );

  root.directory('source/_posts', dir => {
    dato.episodes.sort(by('id', 'episodeNumber')).forEach(episode => {
      dir.createPost(
        `${episode.slug()}.md`,
        'yaml',
        {
          frontmatter: {
            title:         episode.title,
            episodeNumber: episode.episodeNumber,
            paletteUrl:    episode.image && episode.image.url({ auto: 'enhance', palette: 'json' }),
            imageUrl:      episode.image && episode.image.url({ w: 500 }),
            thumbnailUrl:  episode.image && episode.image.url({ w: 500, h: 280, fit: 'crop', auto: 'enhance', fm: 'jpg' }),
            date:          episode.firstAired.toMap(),
            rating:        episode.rating,
            director:      episode.director,
            categories:    [episode.season.name]
          },
          content: episode.description
        }
      );
    });
  });

  const characters = dato.characters.sort(by('position'));

  root.createDataFile('source/_data/characters.yml', 'yaml',
    characters.map(character => {
      return {
        slug:         character.slug({ prefixWithId: false }),
        title:        character.name,
        actorName:    character.actorName,
        thumbnailUrl: character.image.url({ fit: 'crop', crop: 'faces', w: 200, h: 200 }),
      };
    })
  );

  root.directory('source/character/', dir => {
    characters.forEach(character => {
      dir.createPost(
        `${character.slug({ prefixWithId: false })}/index.md`,
        'yaml',
        {
          frontmatter: {
            layout:       'character',
            title:        character.name,
            actorName:    character.actorName,
            episodes:     parseInt(character.episode),
            imageUrl:     character.image.url({ w: 500, fm: 'jpg' }),
          },
          content: character.description
        }
      );
    });
  });
};
